### ISSUE TITLE
Failed to load the jupyterlab-git server extension

### ISSUE THREAD

**Issue by GretaVilla** [jupyterlab/jupyterlab-git#1237]
I'm using an Azure virtual machine. In jupyter lab I downloaded the git extension with:
`pip install jupyterlab-git `
and downloaded @jupyterlab/git from the extension manager.

After refresh I get the following error:
"Git server extension is unavailable. Please ensure you have installed the JupyterLab Git server extension by running: pip install --upgrade jupyterlab-git. To confirm that the server extension is installed, run: jupyter server extension list."

![image](https://user-images.githubusercontent.com/103126089/234547098-e119fbce-cf4a-4398-b022-63051a672d9c.png)

I did run `pip install --upgrade jupyterlab-git ` and everything seems fine, requirements were already installed.
From the output of jupyter server extension list I can correctly see the extension:

![image](https://user-images.githubusercontent.com/103126089/234548819-fe031dd9-729a-43c2-a251-327f198a1873.png)![image](https://user-images.githubusercontent.com/103126089/234548249-daab732c-5b0f-46bc-ba03-0ac71b182ad0.png)

**Comment by HananInfiniGrow** [comment]
I have the same issue. Both front and backend version is v0.41.0

```
(hub) ubuntu@ip-172-31-31-148:/opt/tljh/hub$ sudo jupyter server extension list
Config dir: /root/.jupyter

Config dir: /usr/etc/jupyter

Config dir: /usr/local/etc/jupyter
    jupyter_server_fileid enabled
    - Validating jupyter_server_fileid...
Package jupyter_server_fileid took 0.0014s to import
      jupyter_server_fileid 0.9.0 OK
    jupyter_server_mathjax enabled
    - Validating jupyter_server_mathjax...
Package jupyter_server_mathjax took 0.0006s to import
      jupyter_server_mathjax  OK
    jupyter_server_terminals enabled
    - Validating jupyter_server_terminals...
Package jupyter_server_terminals took 0.0028s to import
      jupyter_server_terminals 0.4.4 OK
    jupyter_server_ydoc enabled
    - Validating jupyter_server_ydoc...
Package jupyter_server_ydoc took 0.0142s to import
      jupyter_server_ydoc 0.8.0 OK
    jupyterlab enabled
    - Validating jupyterlab...
Package jupyterlab took 0.0361s to import
      jupyterlab 3.6.3 OK
    jupyterlab_git enabled
    - Validating jupyterlab_git...
Package jupyterlab_git took 0.0158s to import
      jupyterlab_git 0.41.0 OK
    nbclassic enabled
    - Validating nbclassic...
Package nbclassic took 0.0000s to import
A `_jupyter_server_extension_points` function was not found in nbclassic. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
      nbclassic 0.5.6 OK
    nbdime enabled
    - Validating nbdime...
Package nbdime took 0.0000s to import
      nbdime 3.2.0 OK
    notebook_shim enabled
    - Validating notebook_shim...
Package notebook_shim took 0.0000s to import
A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
      notebook_shim  OK
```
```
JupyterLab v3.6.3
/usr/local/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        nbdime-jupyterlab v2.2.0 enabled OK
        @jupyterlab/git v0.41.0 enabled OK (python, jupyterlab-git)

Other labextensions (built into JupyterLab)
   app dir: /usr/local/share/jupyter/lab
```

**Comment by fcollonval** [comment]
Could you please execute JupyterLab in debug mode (CLI option `--debug`) and provide here the server logs?

**Comment by HananInfiniGrow** [comment]
Since I am using TLJH I don't run jupyterlab explicitly. The hub runs it
automatically. I don't know how to turn on debugging in this situation.
I might just cancel the TLJH and go with a simple bare metal jupyterlab.

**Comment by GretaVilla** [comment]
I'm running jupyterLab from an Azure ML compute instance, which automatically opens JupyterLab.  I'm not sure how to turn on debugging

**Comment by HYBG-1126** [comment]
I have the same issue，but i don't know what should i do;

**Comment by mk-raven** [comment]
have the same problem with installl extensions in jupyterhub.
Any update?

**Comment by krassowski** [comment]
No one provided the jupyter server logs with `--debug` so for now the only thing we can say is that it is likely because the extension was installed after the server has already started.

**Comment by krassowski** [comment]
Ah, there are logs above after all. In JupyterHub the problem might be because extension was installed after the server started. No idea as to AWS SageMaker - maybe best to ask SageMaker folks?

CC @JasonWeill in case if you know what the procedure for reporting issues specific to AWS SageMaker might be.

**Comment by mk-raven** [comment]
> No one provided the jupyter server logs with `--debug` so for now the only thing we can say is that it is likely because the extension was installed after the server has already started.

is it problem if extension installed after server was running?
Maybe could be some solution?

**Comment by fcollonval** [comment]
> is it problem if extension installed after server was running?

Yes because there are no dynamic loading of extensions. So if it not there before, it will never appear.

> Maybe could be some solution?

No magic except restarting the server to reload extensions.

**Comment by arpitgaur02** [comment]
I got the same error when I installed the git extension for the first time and refreshed the jupyter lab, but when I relaunch the lab there was no such issue.

**Comment by evansmn** [comment]
I would appreciate a clear start-to-finish set of commands to get this working.  I have tried (debian linux) as both regular user and admin jupyter-user in a terminal using sudo -E: 
* stop the jupyterhub service
* conda activate venv
* conda install jupyterhub-git
* start the jupyterhub service
* install using pypi into the same venv as jupyter-user.

I get confirmations about the extension being installed (v  0.50.2 ) but always the error: "Failed to load the jupyterlab-git server extension Git server extension is unavailable. Please ensure you have installed the JupyterLab Git server extension by running: pip install --upgrade jupyterlab-git. To confirm that the server extension is installed, run: jupyter server extension list."

**Comment by natema** [comment]
Same issue here, using the automatic installation process of [IJulia](https://github.com/JuliaLang/IJulia.jl).
